### ISSUE TITLE
MAME 0.174+ dumps core with some games on Linux (might be GCC 6-related)

### ISSUE THREAD

**Issue by mdoege** [mamedev/mame#1183]
In MAME 0.174 and later (on Arch Linux 64 bit), some games and machines will cause MAME to crash immediately on startup:

exerion
batmanfr
vectrex
wecleman
sonicpop
smashtv

```
$ LANG=C sdlmame exerion
Value 1 not supported for option sound - falling back to auto
/usr/bin/sdlmame: line 56:  5758 Segmentation fault      (core dumped) ./sdlmame "$@"
```
I have compiled MAME 0.173 under GCC 6 today, and sonicpop and vectrex crashes but everything else works. So this might be a GCC 6-related issue. (Arch Linux switched to GCC 6 between the 0.173 and 0.174 MAME releases.)

The games above work fine on Windows (official 64 bit MAME binary) and OS X (Homebrew), so this is purely a Linux issue.

**Comment by cuavas** [comment]
Can you please try adding `ARCHOPTS=-flifetime-dse=1` to your make command?  GCC6 changes behaviour in a way MAME's allocators to behave differently.

**Comment by startaq** [comment]
I can't reproduce this, gcc (GCC) 6.1.1 20160707 on Arch Linux. Exerion works fine here with latest master.

**Comment by mdoege** [comment]
@cuavas Thanks, your fix has resolved this issue for me. I will file a bug report with the package maintainer.

@startaq Maybe the latest git master has some other fix for this then? I have added the fix by cuavas to the 0.176-2 PKGBUILD but have not tried the latest git version.

**Comment by mdoege** [comment]
@startaq Also, is your system 32 bit or 64 bit?
